aws/table_aws_codestar_notification_rule.go

```diff
@@ -0,0 +1,193 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/codestarnotifications"
+
+	"github.com/turbot/go-kit/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsCodestarNotificationRule(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_codestar_notification_rule",
+		Description: "AWS CodeStar Notification Rule",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("arn"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"NotFound", "InvalidParameter"}),
+			},
+			Hydrate: describeNotificationRule,
+			Tags:    map[string]string{"service": "codestar-notifications", "action": "DescribeNotificationRule"},
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listNotificationRules,
+			Tags:    map[string]string{"service": "codestar-notifications", "action": "ListNotificationRules"},
```

Comment:
Could we add the optional key qualifiers here to minimize the API response and query time? The API [ListNotificationRules](https://docs.aws.amazon.com/codestar-notifications/latest/APIReference/API_ListNotificationRules.html) supports [Filters](https://docs.aws.amazon.com/codestar-notifications/latest/APIReference/API_ListNotificationRules.html#codestarnotifications-ListNotificationRules-request-Filters) as input parameters.

For reference, please see the code at: [Steampipe Plugin AWS: table_aws_ec2_instance.go](https://github.com/turbot/steampipe-plugin-aws/blob/main/aws/table_aws_ec2_instance.go#L541-L546).

Reply:
I've done this and was able to check that I'm still able to filter by the relevant columns, but I haven't been able to verify that this has the intended performance impact.

There are also two possible filters that don't map to any specific columns. Is there anything to be done in this case?

Reply:
> There are also two possible filters that don't map to any specific columns. Is there anything to be done in this case?

We can address this by:

- Adding a column to accept the filter parameter:

```go
{
  Name:        "<Column Name>",
  Description: "...",
  Type:        <Column Type>,
  Transform:   transform.FromQual("<Column Name>"),
},
```

- Using this column in the function [buildCodeStarNotificationRulesFilter](https://github.com/turbot/steampipe-plugin-aws/pull/2198/files#diff-d94afafdc25a86b7a1149cfa7f29faba181fa61c7be19d0d01dc0ef7ad1ad903R214-R217) to build the filter parameter.

Note: This column value will be populated when the value is specified as a query parameter in the where clause.

Reply:
I've tried to do this but didn't get it to work. The following returns zero rows:
```
select
  name, target_address
from
  aws_codestar_notification_rule
where target_address = 'arn:aws:sns:***' ;
```
Whereas the following returns one result:
```
aws codestar-notifications list-notification-rules \
  --filters Name=TARGET_ADDRESS,Value=arn:aws:sns:***
```
I've pushed the code so would appreciate if you can have a look at it.